logstash-core/src/main/java/org/logstash/ackedqueue/AckedBatch.java

```diff
@@ -36,14 +37,11 @@ public static AckedBatch create(Batch batch) {
         return ackedBatch;
     }
 
-    public RubyHash toRubyHash(final Ruby runtime) {
-        final RubyBoolean trueValue = runtime.getTrue();
-        final RubyHash result = RubyHash.newHash(runtime);
-        this.batch.getElements().forEach(e -> result.fastASet(
-            JrubyEventExtLibrary.RubyEvent.newRubyEvent(runtime, (Event) e),
-            trueValue
-            )
-        );
+    public Set<RubyEvent> toSet() {
```

Comment:
I've a doubt about the data structure used, but probably I miss something in full picture. We use LinkedHashSet because order is important, we use Set because we desire to avoid duplication, but every time we generate an Event instance, the instance has it's unique hashcode, so I think we shouldn't have duplication of events in the Batch.
At the end the returned Set is used as a Collection, could we use List/ArrayList instead of Set/LinkedHashSet?

Reply:
Yeah, good question.  The `MemoryReadBatch`  was/is using a `LinkedHashSet` to avoid duplication in the `merge()` method. When I refactored the `AckedReadBatch` I figured I'd keep the same semantic. That provided to possibility to eliminate that `RubyHash` conversion in `AckedBatch` and simply return a `LinkedHashSet` of the dequeued elements to directly create that `Set` in `AcketReadBatch` without using any `RubyHash` in there. 
I will double-check the necessity of using a  Set in there.

Reply:
I decided to refactor that, see comments.